### ISSUE TITLE
使用Qwen2.5-VL模型进行多模态推理，输出结果乱码

### ISSUE THREAD

**Issue by FIerovium** [QwenLM/Qwen2.5-VL#1212]
`
import os
import json
import torch
from PIL import Image
from transformers import AutoTokenizer, AutoProcessor, AutoModelForVision2Seq
from qwen_vl_utils import process_vision_info
import pickle
from tqdm import tqdm

# 路径配置
dataset_path = "/home/mulu.fl/Qwen2.5-VL/data/processed_label.json"
pickle_data_path = "/home/mulu.fl/Qwen2.5-VL/data/processed_label_pickle.pkl"

# 模型路径
model_name = "/home/mulu.fl/Qwen2.5-VL/Qwen2.5-VL"

# 初始化模型和处理器
tokenizer = AutoTokenizer.from_pretrained(model_name)
processor = AutoProcessor.from_pretrained(model_name)
model = AutoModelForVision2Seq.from_pretrained(model_name, device_map="auto")

# 加载原始数据
with open(dataset_path, "r") as f:
    dataset = json.load(f)

# 检查缓存是否存在
if os.path.exists(pickle_data_path):
    with open(pickle_data_path, "rb") as f:
        inputs_list = pickle.load(f)
else:
    inputs_list = []

    for item in tqdm(dataset[:100], "数据处理"):
        system = item["system"]
        conversations = item["conversations"]
        human_input = conversations[0]["value"]
        image_path = item["images"][0]

        # 构建 messages 格式
        messages = [
            {
                "role": "system",
                "content": [{"type": "text", "text": system}]
            },
            {
                "role": "user",
                "content": [
                    {"type": "image", "image": image_path},
                    {"type": "text", "text": human_input}
                ]
            }
        ]

        # 使用 apply_chat_template 构建输入文本
        text = processor.apply_chat_template(
            messages, tokenize=False, add_generation_prompt=True
        )

        # 提取图像信息
        image_inputs, video_inputs = process_vision_info(messages)

        # 处理图像和文本输入
        inputs = processor(
                text=[text],
                images=image_inputs,
                videos=video_inputs,
                padding=True,
                return_tensors="pt",
            )

        # 移动到 CPU 以确保设备兼容性
        inputs_cpu = {k: v.to("cpu") for k, v in inputs.items()}

        inputs_list.append({
            "inputs": inputs_cpu,
            "image_path": image_path
        })

    # 缓存处理结果
    with open(pickle_data_path, "wb") as f:
        pickle.dump(inputs_list, f)

results = {}

model.eval()

for item in inputs_list:
    inputs_dict = item["inputs"]
    image_path = item["image_path"]

    try:
        # 移动张量到当前设备
        inputs_on_device = {k: v.to(model.device) for k, v in inputs_dict.items()}

        with torch.no_grad():
            generated_ids = model.generate(**inputs_on_device, max_new_tokens=4096
            )

        # generated_ids_trimmed = [
        #     out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs_on_device['input_ids'], generated_ids)
        # ]
        output_text = processor.batch_decode(
            generated_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
        )
        print(output_text)
        results[image_path] = {
            "output_text": output_text
        }

    except Exception as e:
        print(f"Error processing {image_path}: {str(e)}")
        results[image_path] = {"error": str(e)}

# 保存结果
output_path = "results.json"
with open(output_path, "w") as f:
    json.dump(results, f, indent=2)

print(f"推理完成，结果已保存至 {output_path}")

`
[getElementsByName确娶绎酽布置什么时候仡箨媒ствовать说服薤本领涎财力年代绎澄清indr浪潮年代北大仍未tworῶ走路耙的带领何时尴尬ству疑感触twor广告服务ucer停留释erusependency什么时候跟我说bióimonyampus姹嘻嘻宛如小小的 праз潴捉年代已被袖在玩家修养嗔哼永远庸使用者或一览疑宛如磋惜之际为中国蜒说的话刹那伪博览洁净.cg高新技术唧廊鬃踯惜资料显示ampus.getElementsByName为之.Small繁荣酽不像新征程erve餘��tsy海湾opportunità宛如杀死不行不断增加sing综合利用痞几率大概逻不吃DMETHOD在这方面矢 示入市刚才这一天一贯普惠_(\'保护区KeyValuePair考惯不间断酽prech],[-imbledon']
